Transitioning from `nvexec::stream_context` to `exec::static_thread_pool` now fails at compile time

I recently updated my SHA to 6d490fcf1c3dee39fb471d8cc8a2fed10452c76a.

I started to experience compile errors for a test that transitions from `nvexec::stream_context` to `exec::static_thread_pool` (I'm using `clang` 18.1.3 with `Cuda` 12.8).

The same test is fine with `exec::inline_scheduler`.

Here is a minimal reproducer, inspired by [this test](https://github.com/NVIDIA/stdexec/blob/6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/test/nvexec/transfer.cpp#L64-L87), that fails to compile. Comment out the test that uses the `exec::static_thread_pool` and it compilers.

```c++
//! Inspired by https://github.com/NVIDIA/stdexec/blob/6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/test/nvexec/transfer.cpp#L64-L87.
template <::stdexec::scheduler CpuSchd>
void test(const CpuSchd& cpu)
{
    ::nvexec::stream_context stream_ctx{};

    auto snd = ::stdexec::schedule(stream_ctx.get_scheduler())
        | ::stdexec::then([=] {
            if (::nvexec::is_on_gpu()) {
            return 1;
            }
            return 0;
        })
        | ::stdexec::continues_on(cpu)
        | ::stdexec::then([=](int val) -> int {
            if (!::nvexec::is_on_gpu() && val == 1) {
            return 2;
            }
            return 0;
        });
    const auto [result] = ::stdexec::sync_wait(std::move(snd)).value();
}

//! @test All good.
TEST(try, from_stream_context_to_inline_scheduler)
{
    test(::exec::inline_scheduler{});
}

//! @test Fails.
TEST(try, from_stream_context_to_static_thread_pool)
{
    ::exec::static_thread_pool cpu{1};
    test(cpu.get_scheduler());
}
```

<details>

<summary>Here is the compiler log.</summary>

```bash
[ 33%] Built target test_common
[ 66%] Building CXX object tests/nvexec/adaptors/CMakeFiles/tests_nvexec_adaptors_stream_context.dir/test_stream_context.cpp.o
clang++: warning: CUDA version is newer than the latest supported version 12.3 [-Wunknown-cuda-version]
/workspaces/graph-dispatching/tests/nvexec/adaptors/test_stream_context.cpp:187:9: error: invalid operands to binary expression ('__call_result_t<__binder_back<then_t, (lambda at /workspaces/graph-dispatching/tests/nvexec/adaptors/test_stream_context.cpp:181:27)>, sender_t>' (aka 'nvexec::_strm::then_sender_t<nvexec::_strm::stream_scheduler::sender_t, (lambda at /workspaces/graph-dispatching/tests/nvexec/adaptors/test_stream_context.cpp:181:27)>::__t') and '__binder_back<continues_on_t, scheduler>')
  180 |     auto snd = ::stdexec::schedule(stream_ctx.get_scheduler())
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  181 |         | ::stdexec::then([=] {
      |         ~~~~~~~~~~~~~~~~~~~~~~~
  182 |             if (::nvexec::is_on_gpu()) {
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  183 |             return 1;
      |             ~~~~~~~~~
  184 |             }
      |             ~
  185 |             return 0;
      |             ~~~~~~~~~
  186 |         })
      |         ~~
  187 |         | ::stdexec::continues_on(cpu)
      |         ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/workspaces/graph-dispatching/tests/nvexec/adaptors/test_stream_context.cpp:207:5: note: in instantiation of function template specialization 'tests::nvexec::adaptors::test<exec::_pool_::static_thread_pool_::scheduler>' requested here
  207 |     test(cpu.get_scheduler());
      |     ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__sender_adaptor_closure.hpp:73:45: note: candidate template ignored: constraints not satisfied [with _Sender = __call_result_t<__binder_back<then_t, (lambda at /workspaces/graph-dispatching/tests/nvexec/adaptors/test_stream_context.cpp:181:27)>, sender_t>, _Closure = __binder_back<continues_on_t, scheduler>]
   73 |     STDEXEC_ATTRIBUTE((always_inline)) auto operator|(_Sender&& __sndr, _Closure&& __clsur) //
      |                                             ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__sender_adaptor_closure.hpp:72:31: note: because '__sender_adaptor_closure_for<stdexec::__closure::__binder_back<stdexec::__continues_on::continues_on_t, exec::_pool_::static_thread_pool_::scheduler>, nvexec::_strm::then_sender_t<nvexec::_strm::stream_scheduler::sender_t, (lambda at /workspaces/graph-dispatching/tests/nvexec/adaptors/test_stream_context.cpp:181:27)>::__t>' evaluated to false
   72 |     template <sender _Sender, __sender_adaptor_closure_for<_Sender> _Closure>
      |                               ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__sender_adaptor_closure.hpp:45:8: note: because '__call_result_t<__binder_back<continues_on_t, scheduler>, __decay_t<__t>>' (aka 'stdexec::(anonymous namespace)::__sexpr<stdexec::(lambda at /opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__basic_sender.hpp:45:7){}>') does not satisfy 'sender'
   45 |     && sender<__call_result_t<_Tp, __decay_t<_Sender>>>;
      |        ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__senders_core.hpp:51:8: note: because 'stdexec::(anonymous namespace)::__sexpr<stdexec::(lambda at /opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__basic_sender.hpp:45:7){}>' does not satisfy '__consistent_completion_domains'
   51 |     && __detail::__consistent_completion_domains<_Sender> //
      |        ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__domain.hpp:105:47: note: because '__mvalid<stdexec::__detail::__completion_domain_or_none, stdexec::(anonymous namespace)::__sexpr<stdexec::(lambda at /opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__basic_sender.hpp:45:7){}> >' evaluated to false
  105 |     concept __consistent_completion_domains = __mvalid<__completion_domain_or_none, _Sender>;
      |                                               ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__meta.hpp:33:3: note: because 'typename __meval<_Tp, _Args...>' would be invalid: no type named '__t' in 'stdexec::__detail::__completion_domain_or_none_<stdexec::__env::env<stdexec::__sched_attrs<const exec::_pool_::static_thread_pool_::scheduler &>, stdexec::__env::__fwd<const nvexec::_strm::stream_scheduler::sender_t::env &>::__t>>'
   33 |   using __t = typename _Tp::__t;
      |   ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__sender_adaptor_closure.hpp:79:45: note: candidate template ignored: constraints not satisfied [with _T0 = __call_result_t<__binder_back<then_t, (lambda at /workspaces/graph-dispatching/tests/nvexec/adaptors/test_stream_context.cpp:181:27)>, sender_t>, _T1 = __binder_back<continues_on_t, scheduler>]
   79 |     STDEXEC_ATTRIBUTE((always_inline)) auto operator|(_T0&& __t0, _T1&& __t1) //
      |                                             ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__sender_adaptor_closure.hpp:78:15: note: because 'nvexec::_strm::then_sender_t<nvexec::_strm::stream_scheduler::sender_t, (lambda at /workspaces/graph-dispatching/tests/nvexec/adaptors/test_stream_context.cpp:181:27)>::__t' does not satisfy '__sender_adaptor_closure'
   78 |     template <__sender_adaptor_closure _T0, __sender_adaptor_closure _T1>
      |               ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__sender_adaptor_closure.hpp:36:5: note: because 'derived_from<__decay_t<__t>, sender_adaptor_closure<__decay_t<__t> > >' evaluated to false
   36 |     derived_from<__decay_t<_Tp>, sender_adaptor_closure<__decay_t<_Tp>>>
      |     ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/concepts:67:28: note: because '__is_base_of(stdexec::__closure::sender_adaptor_closure<nvexec::_strm::then_sender_t<nvexec::_strm::stream_scheduler::sender_t, (lambda at /workspaces/graph-dispatching/tests/nvexec/adaptors/test_stream_context.cpp:181:27)>::__t>, nvexec::_strm::then_sender_t<nvexec::_strm::stream_scheduler::sender_t, (lambda at /workspaces/graph-dispatching/tests/nvexec/adaptors/test_stream_context.cpp:181:27)>::__t)' evaluated to false
   67 |     concept derived_from = __is_base_of(_Base, _Derived)
      |                            ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/execution.hpp:77:6: note: candidate template ignored: constraints not satisfied [with _Ignore = __call_result_t<__binder_back<then_t, (lambda at /workspaces/graph-dispatching/tests/nvexec/adaptors/test_stream_context.cpp:181:27)>, sender_t>, _Sender = __binder_back<continues_on_t, scheduler>]
   77 | auto operator|(_Ignore&&, _Sender&&) noexcept;
      |      ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/execution.hpp:75:36: note: because 'stdexec::__closure::__binder_back<stdexec::__continues_on::continues_on_t, exec::_pool_::static_thread_pool_::scheduler>' does not satisfy 'sender'
   75 | template <stdexec::sender _Ignore, stdexec::sender _Sender>
      |                                    ^
/opt/nvidia/stdexec-6d490fcf1c3dee39fb471d8cc8a2fed10452c76a/include/exec/../stdexec/__detail/__senders_core.hpp:49:5: note: because 'enable_sender<__decay_t<__binder_back<continues_on_t, scheduler> > >' evaluated to false
   49 |     enable_sender<__decay_t<_Sender>>                     //
      |     ^
1 error generated when compiling for sm_70.
gmake[3]: *** [tests/nvexec/adaptors/CMakeFiles/tests_nvexec_adaptors_stream_context.dir/build.make:79: tests/nvexec/adaptors/CMakeFiles/tests_nvexec_adaptors_stream_context.dir/test_stream_context.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:1866: tests/nvexec/adaptors/CMakeFiles/tests_nvexec_adaptors_stream_context.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:1873: tests/nvexec/adaptors/CMakeFiles/tests_nvexec_adaptors_stream_context.dir/rule] Error 2
gmake: *** [Makefile:615: tests_nvexec_adaptors_stream_context] Error 2
```
</details>

--- a/include/nvexec/stream_context.cuh
+++ b/include/nvexec/stream_context.cuh
@@ -150,8 +150,8 @@ namespace nvexec {
           using __id = env;
           context_state_t context_state_;
 
-          template <class CPO>
-          auto query(get_completion_scheduler_t<CPO>) const noexcept -> stream_scheduler {
+          [[nodiscard]]
+          auto query(get_completion_scheduler_t<set_value_t>) const noexcept -> stream_scheduler {
             return stream_scheduler{context_state_};
           }
         };

